benchmark/src/main/java/org/corfudb/benchmark/ParseArgs.java

```diff
@@ -0,0 +1,123 @@
+
+    @Getter
+    protected int numStreams;
+    /**
+     * Server endpoint.
+     */
+    @Getter
+    @NonNull
+    protected String endpoint;
+
+    /**
+     * ratio of CorfuTable put operation.
+     */
+    @Getter
+    protected double ratio;
+
+    /**
+     * operation name.
+     */
+    @Getter
+    @NonNull
+    protected String op;
+
+    @Getter
+    protected int keyNum;
+
+    @Getter
+    protected int valueSize;
+
+    ParseArgs(String[] args) {
+        Args cmdArgs = new Args();
+        JCommander jc = JCommander.newBuilder()
+                .addObject(cmdArgs)
+                .build();
+        jc.parse(args);
+
+        if (cmdArgs.help) {
+            jc.usage();
+            System.exit(0);
+        }
+        numRequests = cmdArgs.numRequests;
+        numRuntimes = cmdArgs.numClients;
+        numThreads = cmdArgs.numThreads;
+        numStreams = cmdArgs.numStreams;
+        endpoint = cmdArgs.endpoint;
+        ratio = cmdArgs.ratio;
+        op = cmdArgs.op;
+        keyNum = cmdArgs.keyNum;
+        valueSize = cmdArgs.valueSize;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("ParseArgs: numRuntimes=").append(numRuntimes)
+                .append(", numThreads=").append(numThreads)
+                .append(", numStreams=").append(numStreams)
+                .append(", numRequests=").append(numRequests)
+                .append(", endpoint=").append(endpoint)
+                .append(", ratio=").append(ratio)
+                .append(", operation=").append(op)
+                .append(", keyNum=").append(keyNum)
+                .append(", valueSize=").append(valueSize);
+        return stringBuilder.toString();
+    }
+
+    public static class Args {
+        @Parameter(names = {"-h", "--help"}, description = "Help message", help = true)
+        boolean help;
+
+        @Parameter(names = {"--endpoint"}, description = "Cluster endpoint", required = true)
+        String endpoint; //ip:portnum
+
+        @Parameter(names = {"--num-clients"}, description = "Number of clients", required = true)
+        int numClients;
+
+        @Parameter(names = {"--num-threads"}, description = "Total number of threads", required = true)
+        int numThreads;
+
+        @Parameter(names = {"--num-requests"}, description = "Number of requests per thread", required = true)
+        int numRequests;
+
+        @Parameter(names = {"--num-streams"}, description = "Number of streams", required = true)
+        int numStreams;
+
+        @Parameter(names = {"--ratio"}, description = "num of put operations / num of requests", required = false)
+        double ratio;
+
+        @Parameter(names = {"--op"}, description = "operation you want to test", required = true)
+        String op;
+
+        @Parameter(names = {"--key-num"}, description = "maximum number of keys you want to store", required = false)
+        int keyNum;
```

Comment:
default value is 0 and will have IllegalArgumentException when generateKey with random

Reply:
Can you explain more?

Reply:
Since it is not required, the value would be 0 when user doesn't input this argument. In this case, IllegalArgumentException will be threw in `int index = random.nextInt(this.keyNum);`